mmrazor/models/architectures/backbones/searchable_autoformer.py

```diff
@@ -0,0 +1,368 @@
+            self.attn,
+            self.mutable_q_embed_dims,
+            True,
+            end=self.mutable_q_embed_dims.current_choice)
+        MutableChannelContainer.register_mutable_channel_to_module(
+            self.attn.rel_pos_embed_k, self.mutable_head_dims, False)
+        MutableChannelContainer.register_mutable_channel_to_module(
+            self.attn.rel_pos_embed_v, self.mutable_head_dims, False)
+
+        # handle the mutable of fc
+        MutableChannelContainer.register_mutable_channel_to_module(
+            self.fc1, mutable_embed_dims, False)
+        MutableChannelContainer.register_mutable_channel_to_module(
+            self.fc1,
+            self.middle_channels,
+            True,
+            start=0,
+            end=self.middle_channels.current_choice)
+        MutableChannelContainer.register_mutable_channel_to_module(
+            self.fc2,
+            self.middle_channels,
+            False,
+            start=0,
+            end=self.middle_channels.current_choice)
+        MutableChannelContainer.register_mutable_channel_to_module(
+            self.fc2, mutable_embed_dims, True)
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        """Forward of Transformer Encode Layer."""
+        residual = x
+        x = self.norm1(x)
+        x = self.attn(x)
+        x = residual + x
+        residual = x
+        x = self.norm2(x)
+        x = self.fc1(x)
+        x = self.act(x)
+        x = self.fc2(x)
+        return residual + x
+
+
+@MODELS.register_module()
+class AutoformerBackbone(BaseBackbone):
+    """Autoformer backbone.
+
+    A PyTorch implementation of Autoformer introduced by:
+    `AutoFormer: Searching Transformers for Visual Recognition
+    <https://arxiv.org/abs/2107.00651>`_
+
+    Modified from the `official repo
+    <https://github.com/microsoft/Cream/blob/main/AutoFormer/>`.
+
+    Args:
+        img_size (int, optional): The image size of input.
+            Defaults to 224.
+        patch_size (int, optional): The patch size of autoformer.
+            Defaults to 16.
+        in_channels (int, optional): The input channel dimension.
+            Defaults to 3.
+        drop_rate (float): Probability of an element to be zeroed.
+            Defaults to 0.
+        drop_path_rate (float): stochastic depth rate. Defaults to 0.
+        qkv_bias (bool, optional): Whether to keep bias of qkv.
+            Defaults to True.
+        norm_cfg (Dict, optional): The config of normalization.
+            Defaults to dict(type='mmrazor.DynamicLayerNorm').
+        act_cfg (Dict, optional): The config of activation functions.
+            Defaults to dict(type='GELU').
+        use_final_norm (bool, optional): Whether use final normalization.
+            Defaults to True.
+        init_cfg (Dict, optional): The config for initialization.
+            Defaults to None.
+    """
+    mutable_settings: Dict[str, List] = {
```

Comment:
make `mutable_settings` an input args for `AutoformerBackbone`, unifying the representation of search space